### PR TITLE
[None][perf] Precompute GDN cache reset indices once per forward

### PR DESCRIPTION
**Pull request by amukkara**
@coderabbitai summary

## Description

TTFT improvement

 Qwen3.5-35B-A3B-FP8, 1 H100 400W, BS=1, ISL=10K / OSL=1
  
  | Config | Before | After | Δ |
  |--------|-------:|------:|---:|
  | piecewise = off | 268.7 ms | 263.0 ms | 2.1% |
  | piecewise = on | 282.7 ms | 278.0 ms | 1.7% |

## Test Coverage

Existing Qwen3.5 accuracy tests in `tests/integration/defs/accuracy/test_llm_api_pytorch.py::TestQwen3_5_35B_A3B`

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: tensorrt_llm/_torch/modules/mamba/gdn_mixer.py [hunk 1]
```diff
@@ -1036,14 +1036,12 @@ def forward_core(
...
         if num_prefills > 0:
-            # PyExecutor guarantees prefill requests are placed before decode requests
-            has_initial_states_p = has_initial_states[:num_prefills]
-            ssm_states[state_indices_p[~has_initial_states_p]] = torch.zeros(
-                (), dtype=ssm_states.dtype, device=ssm_states.device
-            )
-            conv_states[state_indices_p[~has_initial_states_p]] = torch.zeros(
-                (), dtype=conv_states.dtype, device=conv_states.device
-            )
+            # Zero conv/ssm cache slots for prefill sequences with no initial
```
Comment (nv-guomingz) [inline tensorrt_llm/_torch/modules/mamba/gdn_mixer.py:1039]: @liji-nv  just filed one PR for fixing PCG perf for qwen3.x models. https://github.com/NVIDIA/TensorRT-LLM/pull/16716 .
Could u plz take a look his pr first?

Comment (amukkara) [reply tensorrt_llm/_torch/modules/mamba/gdn_mixer.py:1039]: thanks for pointing that out. with 16716, this PR is redundant.

**Comment by amukkara** [comment]
Closing since #16716 fixes the same issue.
